pkg/controller/dnszone/test_helpers.go

```diff
@@ -49,6 +51,9 @@ var (
 						},
 					},
 				},
+				Azure: &hivev1.AzureDNSZoneSpec{
```

Comment:
This does not make a valid DNSZone. A DNSZone cannot be for both AWS and Azure.

Reply:
Created separate dns zone objects for azure